src/os/file.go

```diff
@@ -170,13 +172,35 @@ func (f *File) WriteString(s string) (n int, err error) {
 	return f.Write([]byte(s))
 }
 
-func (f *File) WriteAt(b []byte, off int64) (n int, err error) {
+var errWriteAtInAppendMode = errors.New("os: invalid use of WriteAt on file opened with O_APPEND")
+
+// WriteAt writes len(b) bytes to the File starting at byte offset off.
+// It returns the number of bytes written and an error, if any.
+// WriteAt returns a non-nil error when n != len(b).
+//
+// If file was opened with the O_APPEND flag, WriteAt returns an error.
+func (f *File) WriteAt(b []byte, offset int64) (n int, err error) {
+	if offset < 0 {
```

Comment:
Perhaps this would be easier to read as:

```go
switch {
    case offset < 0:
        return 0, &PathError{Op: "writeat", Path: f.name, Err: errNegativeOffset}
    case f.handle == nil:
        return 0, &PathError{Op: "writeat", Path: f.name, Err: ErrClosed}
    case f.appendMode:
        return 0, errWriteAtInAppendMode
}
```